py/usageaccountant/datadog_fetcher.py

```diff
@@ -0,0 +1,292 @@
+import argparse
+import json
+import logging
+import urllib.parse
+from typing import Any, Dict, List, Optional, Tuple, TypedDict
+from urllib.request import Request, urlopen
+
+from usageaccountant.accumulator import (
+    KafkaConfig,
+    UsageAccumulator,
+    UsageUnit,
+)
+
+# TODO fetch these from env
+headers = {"DD-APPLICATION-KEY": "", "DD-API-KEY": ""}
+
+logger = logging.getLogger("fetcher")
+
+# TODO remove this before prod
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[logging.FileHandler("debug.log"), logging.StreamHandler()],
+)
+
+
+class DatadogResponseUnit(TypedDict):
+    plural: str
+    name: str
+
+
+class DatadogResponseSeries(DatadogResponseUnit):
+    unit: List[DatadogResponseUnit]
+    pointlist: List[List[float]]
+    scope: str
+
+
+class DatadogResponse(DatadogResponseSeries):
+    error: Optional[str]
+    query: str
+    from_date: int
+    to_date: int
+    status: str
+    series: List[DatadogResponseSeries]
+
+
+def is_valid_query(query: str) -> bool:
+    """
+    Validates if query string contains shared_resource_id
+    and app_feature–parameters used in UsageAccountant.record().
+
+    query: Datadog query
+    """
+    return "shared_resource_id" in query and "app_feature" in query
+
+
+def get(query: str, start_time: int, end_time: int) -> Any:
+    """
+    Fetches timeseries data from Datadog API, returns response dict.
+
+    query: Datadog query
+    start_time: Start of the time window in Unix epoch format
+                with second-level precision
+    end_time: End of the time window in Unix epoch format
+              with second-level precision
+    """
+    data = {"query": query, "from": start_time, "to": end_time}
+
+    base_url = "https://api.datadoghq.com/api/v1/query"
+    query_string = urllib.parse.urlencode(data)
+    full_url = f"{base_url}?{query_string}"
+
+    try:
+        request = Request(url=full_url, headers=headers)
+        # context is needed to handle
+        # ssl.SSLCertVerificationError:
+        # [SSL: CERTIFICATE_VERIFY_FAILED] on macOS
+        # TODO remove next three lines and
+        # context parameter from the call to urlopen()
+        import ssl
+
+        context = ssl.SSLContext()
```

Comment:
I am not sure why you need this 

```
data = {"query": "asdasd", "from": 123123123, "to": 123123123}

    base_url = "https://api.datadoghq.com/api/v1/query"
    query_string = urllib.parse.urlencode(data)
    full_url = f"{base_url}?{query_string}"

    try:
        request = Request(url=full_url, headers=headers)
        response = urlopen(request).read()
        print(json.loads(response))
```

This works just fine on my laptop. What issue did you run into ?

Reply:
I get 
`<urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)>` 
without SSL Context.

Reply:
I think your python venv is broken. 
You should really fix it. 

How did you install python and create the virtual environment ?